torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -78,19 +80,21 @@ bool isSimpleMap(Node *node) {
   TensorType* expected_type = node->inputs()[0]->type()->cast<TensorType>();
   if (!expected_type)
     return false;
-  static const auto equal_modulo_strides = [](TensorType* expected, const TypePtr& _actual) {
+  static const auto equal_modulo_strides = [](TensorType* expected, const TypePtr& _actual, bool checkScalarType) {
     TensorType* actual = _actual->cast<TensorType>();
     return actual &&
-           expected->scalarType() == actual->scalarType() &&
+           (!checkScalarType || expected->scalarType() == actual->scalarType()) &&
            expected->device() == actual->device() &&
            expected->sizes() == actual->sizes();
   };
+//Don't check scalar types for type_as and comparison, allFloatIO is doing some checks.
+  bool checkScalarType = ((node->kind() != aten::type_as) && !(torch::jit::isComparison(node)));
   for (Value * val : node->inputs()) {
-    if (!equal_modulo_strides(expected_type, val->type()))
+    if (!equal_modulo_strides(expected_type, val->type(), checkScalarType))
```

Comment:
This is incorrect, because it will allow e.g. `float == int` comparisons, which we normally don't support.

Reply:
On the other hand this is probably not the right place to check this thing. `isSimpleMappable` isn't something that has any requirements on input types.

Reply:
Ultimately, it won't allow such comparisons because input types are checked.